src/circuits/getsamples.jl

```diff
@@ -144,12 +166,23 @@ drawn from the probabilty distribution:
 - `P(σ) = |⟨σ|ψ⟩|²`  :  if `M = ψ is MPS`
 - `P(σ) = ⟨σ|ρ|σ⟩`   :  if `M = ρ is MPO`
 """
-function getsamples!(M::Union{MPS,MPO},nshots::Int; kwargs...)
-  measurements = Matrix{Int64}(undef, nshots, length(M))
-  for n in 1:nshots
-    measurements[n,:] = getsamples!(M; kwargs...)
+function getsamples!(M0::Union{MPS,MPO},nshots::Int; kwargs...)
+  nthreads = Threads.nthreads()
+  data = [Vector{Vector{Int64}}(undef, 0) for _ in 1:nthreads]
+  M = copy(M0)
+  M = orthogonalize!(M,1)
+  
+  Threads.@threads for j in 1:nshots
```

Comment:
@mtfishman Can you double check the multi-threaded loop?

Reply:
Glancing over it, it looks fine. Were there any particular issues you were concerned about?

Reply:
You don't get the expected speedup, we will have to test this in details.